ruby/rspec/tutorial.md

```diff
@@ -0,0 +1,153 @@
+
+## Model specs
+They are straightforward unit tests which tests small parts of the system, they shouldn’t test private methods, just model’s methods and callbacks. These unit tests follows the same rules as services, jobs, mailers and any other ruby classes, except controllers that needs some setup.
+
+```
+describe User do
+  describe '#full_name' do
+    let(:user) { create(:user) }
+
+    it 'returns the correct name' do
+      expect(user.full_name).to eq("#{user.first_name} #{user.last_name}")
+    end
+	end
+
+  describe '.search' do
+    let(:user) { create(:user) }
+
+    context 'with the right term' do
+      it 'returns a user' do
+        expect(User.search(user.name)).to include user
+      end
+    end
+
+    context 'with another term' do
+      it 'returns 0 users' do
+        expect(User.search('whatever')).to be_empty
+      end
+    end
+  end
+end
+```
+
+As you can see, instance methods are described with `# `and class methods with `.` .
+
+## Feature specs
+Feature specs are integration tests that aims to cover all the application flow, it starts when a user does something in the UI (like clicking a button) and finishes when the user sees the response (like an alert). Thus they are written from the user viewpoint, but in some cases you can check the Model/DB if there aren’t a visible response or to make sure the app is working as expected.
+
+We use Capybara for interacting with the browser and PhantomJS (through the ruby driver poltergeist) or webkit to test pages with js. You can take a look at the capybara’s GitHub page [about drivers](https://github.com/teamcapybara/capybara#drivers). Try to avoid js tests because they could lead to some headaches. However, sometimes you will need them. An example of feature test is:
+
+```
+describe 'User creates a task' do
+  before do
+	  visit new_task_path
+  end
+
+  it 'can create a task' do
+    fill_in 'Title', with: 'My new task'
+    fill_in 'Description', with: 'I need to do the laundry'
+    click_button 'Create Task'
+
+    expect(page).to have_content 'The task was created'
+  end
+end
+```
+
+Also, Capybara comes with a built-in DSL for creating descriptive acceptance tests, where the previous code can be rewritten as:
+
+```
+feature 'User creates a task' do
+  background do
+	  visit new_task_path
+  end
+
+  scenario 'can create a task' do
+    fill_in 'Title', with: 'My new task'
+    fill_in 'Description', with: 'I need to do the laundry'
+    click_button 'Create Task'
+
+    expect(page).to have_content 'The task was created'
+  end
+end
+```
+
+Both ways are accepted but the first one is the most used in the company because the other types of testings use the same structure. I recommend checking [this guide](https://robots.thoughtbot.com/write-reliable-asynchronous-integration-tests-with-capybara) that brings some tips at the time of doing feature testing.
+
+
+## Controller specs
```

Comment:
Should we say that we prefer request specs here as well? Or below in the next section is enough?

Reply:
clarifying...